### PR TITLE
Add HostnameDiscoveryEvent

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes a part of #2569 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > UTs pass
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

File: monkey/tests/unit_tests/common/agent_events/test_hostname_discovery_event.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+    assert isinstance(HostnameDiscoveryEvent(**dict_with_tags).tags, frozenset)
+
+
+def test_constructor__extra_fields_forbidden():
+    extra_field_dict = HOSTNAME_DISCOVERY_SIMPLE_DICT.copy()
+    extra_field_dict["extra_field"] = 99  # red balloons
+
+    with pytest.raises(ValueError):
+        HostnameDiscoveryEvent(**extra_field_dict)
```
Comment (VakarisZ) [inline monkey/tests/unit_tests/common/agent_events/test_hostname_discovery_event.py:56]: Neither of these properties are defines by hostname discovery event. This is testing `AbstractAgentEvent` and `InfectionMonkeyModelConfig`. We can test that, but this is not the place

**Review by VakarisZ — CHANGES_REQUESTED** [review]
small changes

File: monkey/common/agent_events/hostname_discovery_event.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+from . import AbstractAgentEvent
+
+
+class HostnameDiscoveryEvent(AbstractAgentEvent):
+    """
+    An event that occurs when the Agent identifies the hostname of the machine
+    on which it is running
```
Comment (VakarisZ) [inline monkey/common/agent_events/hostname_discovery_event.py:7]: Doesn't matter if the agent discovered the hostname of the current machine or remote. `AbstractAgentEvent` has `target`. The hostname is of the target machine.

Comment (shreyamalviya) [reply monkey/common/agent_events/hostname_discovery_event.py:7]: `target` is `None` if it is the current machine.

File: monkey/tests/unit_tests/common/agent_events/test_hostname_discovery_event.py (hunk 1 above)
Comment (shreyamalviya) [reply monkey/tests/unit_tests/common/agent_events/test_hostname_discovery_event.py:56]: They test that the event inherits `AbstractAgentEvent` and the right model config.

File: monkey/common/agent_events/hostname_discovery_event.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/common/agent_events/hostname_discovery_event.py:7]: Maybe we should change the name of target to `remote_target` to clarify

File: monkey/tests/unit_tests/common/agent_events/test_hostname_discovery_event.py (hunk 1 above)
Comment (VakarisZ) [reply monkey/tests/unit_tests/common/agent_events/test_hostname_discovery_event.py:56]: You can test that with `issubclass(HostnameDiscoveryEvent, AbstractAgentEvent)`, but I'm not sure we should add these everywhere.

File: monkey/common/agent_events/hostname_discovery_event.py (hunk 2 above)
Comment (mssalvatore) [reply monkey/common/agent_events/hostname_discovery_event.py:7]: We can talk over webex. One way or another, its outside the scope of this PR.

**Comment by mssalvatore** [comment]
Merged by 4325de388e

**Pull request closed without merging**